### PR TITLE
Rename ActorContext#schedule with scheduleOnce

### PR DESCRIPTION
**Pull request by dwijnand**
Deprecate the former

### CODE REVIEW

**Comment by dwijnand** [comment]
Self to-do: check usages and replace, to avoid deprecation messages.

File: akka-actor-typed/src/main/scala/akka/actor/typed/javadsl/ActorContext.scala [hunk 1]
```diff
@@ -207,6 +207,10 @@ trait ActorContext[T] extends akka.actor.typed.ActorContext[T] {
    * This method is thread-safe and can be called from other threads than the ordinary
    * actor message processing thread, such as [[java.util.concurrent.CompletionStage]] callbacks.
    */
+  def scheduleOnce[U](delay: Duration, target: ActorRef[U], msg: U): akka.actor.Cancellable
+
+  @Deprecated
+  @deprecated("Use scheduleOnce", since = "2.5.18")
```
Comment (chbatey) [inline akka-actor-typed/src/main/scala/akka/actor/typed/javadsl/ActorContext.scala]: As this is typed you can just remove and we'll add a note to the release notes

**Review by chbatey — APPROVED** [review]
LGTM

File: akka-actor-typed/src/main/scala/akka/actor/typed/javadsl/ActorContext.scala (hunk 1 above)
Comment (chbatey) [reply akka-actor-typed/src/main/scala/akka/actor/typed/javadsl/ActorContext.scala]: Or make sure that the docs don't include any schedule

Comment (dwijnand) [reply akka-actor-typed/src/main/scala/akka/actor/typed/javadsl/ActorContext.scala]: OK, sure.

**Review by patriknw — COMMENTED** [review]
looking good, after removal of deprecated

**Review by raboof — APPROVED** [review]

**Pull request merged**